tests/test_flow/fakes.py

```diff
@@ -166,8 +169,40 @@ def rm(self, url, recursive=False):
             del self._files_by_url[url]
 
 
+def move_files_in_dir(src_dir, dst_dir):
+    """
+    Move all files from one directory to another.
+    """
+    filenames = os.listdir(src_dir)
+    for filename in filenames:
+        shutil.move(os.path.join(src_dir, filename), dst_dir)
+
+
+@contextmanager
+def temporary_clean_dir(target_dir):
+    """
+    Temporarily remove all files in a directory for the duration of a context.
+    Once the context ends, the directory is restored to its original state. That
+    means any new file created during the duration of the context are deleted.
+    """
+    with TemporaryDirectory() as tmp_backup_dir:
+        try:
+            move_files_in_dir(target_dir, tmp_backup_dir)
+            yield
+        finally:
+            filenames = os.listdir(target_dir)
+            for filename in filenames:
+                path = os.path.join(target_dir, filename)
+                if os.path.isdir(path):
+                    shutil.rmtree(path)
+                else:
+                    os.remove(path)
+
+            move_files_in_dir(tmp_backup_dir, target_dir)
+
+
 @contextmanager
-def run_in_fake_gcp(fake_gcs_fs: FakeGcsFs, caplog):
+def run_in_fake_gcp(fake_gcs_fs: FakeGcsFs, caplog, tmp_path):
```

Comment:
Do we need to remove all files from the `tmp_path` or only from the cache dir? If it's the later, then wdyt about passing the cache dir here instead?

Reply:
Updated to use cache dir.